### DISCUSSION TITLE
Segmentation fault

### DISCUSSION THREAD

**Discussion by albenstein** [su2code/SU2#2101] [General]
*If this is a general question or difficulty building/using the code, please open a "Discussion" instead.*

**Describe the bug**
*Give a clear and concise description of what the bug is and what you expect the behavior to be instead. Please provide enough instructions and data for a developer to reproduce the problem (config, mesh, commands, etc.). Please provide the smallest working example that reproduces the problem.*

**Bug report checklist**
*Please make sure that you have followed the checklist below, many common problems can be solved by:*

- [x] Consulting the build instructions (https://su2code.github.io/docs_v7/Build-SU2-Linux-MacOS/).
- [x] Looking for similar problems on GitHub or CFD-Online (https://www.cfd-online.com/Forums/su2/).
- [x] Updating to the newest version of SU2 (either develop, master, or the pre-built executables https://su2code.github.io/download.html).

**Desktop (please complete the following information):**
 - OS: [e.g., Linux (Redhat 7.9)]
 - C++ compiler and version: [e.g., g++ (GCC) 8.5.0]
 - MPI implementation and version: [e.g., OpenMPI 4.0.3]
 - SU2 Version: [e.g., v7.5.1]

When I use SU2 with QuickStart config file with OpenMPI, it gives error like in attached file.
Command is:
`mpirun -np 8 --allow-run-as-root SU2_CFD inv_NACA0012.cfg`

[segmentation_fault.txt](https://github.com/su2code/SU2/files/12170716/segmentation_fault.txt)

**Comment by kursatyurt** [comment]
This sound like your OpenMPI installation is broken, did you compiled it by yourself?

**Comment by albenstein** [comment]
@kursatyurt Yes, I did.

**Comment by kursatyurt** [comment]
> @kursatyurt Yes, I did.

If you are using a cluster, try to compile with packages installed by the system administrator (most clusters offer modules). If not a local computer, try system packages.

**Reply by albenstein** [reply]
I'm the system administrator and using modules also.

**Reply by kursatyurt** [reply]
Could you try some simple example like following:

```C
#include <stdio.h>
#include "mpi.h”
int main(int argc, char *argv[])
{
 MPI_File fh;
 int buf[1000], rank;
 MPI_Init(0,0);
 MPI_Comm_rank(MPI_COMM_WORLD, &rank);
 MPI_File_open(MPI_COMM_WORLD, "test.out",
 MPI_MODE_CREATE|MPI_MODE_WRONLY,
 MPI_INFO_NULL, &fh);
 if (rank == 0){
 MPI_File_write(fh, buf, 1000, MPI_INT, MPI_STATUS_IGNORE);}
 MPI_File_close(&fh);
 MPI_Finalize();
 return 0;
}
```

Does this works? If you are the administrator did you compiled openmpi with tests? Is it passing the all the tests?

**Reply by albenstein** [reply]
I did not compile with tests, how can I do that?
Also howto compile example?

**Reply by kursatyurt** [reply]
The official guide for validation is https://docs.open-mpi.org/en/v5.0.x/validate.html#

Also, how to build an app with mpi is described here https://docs.open-mpi.org/en/v5.0.x/building-apps/index.html

Just looking at GitHub, I found this test suite https://github.com/open-mpi/mpi-test-suite it might work testing. 

We can investigate other possibilities if your MPI installation can pass the MPI I/O test.

**Reply by albenstein** [reply]
I've just tested hello_c.c example from OpenMPI source directory:

`[root@headnode examples]# mpicc hello_c.c -o hello_c -g`
`[root@headnode examples]# ./hello_c`
`Hello, world, I am 0 of 1, (Open MPI v4.0.3rc4, package: Open MPI root@c-141-67-80-081 Distribution, ident: 4.0.3rc4, repo rev: v4.0.3rc3-36-gfaa1bdc, Feb 27, 2020, 136)`

**Reply by albenstein** [reply]
I've tried to use https://github.com/open-mpi/mpi-test-suite, but it gave compiling error.

**Reply by kursatyurt** [reply]
I'm sorry, but running a hello world example on one process doesn't show anything. You have to test MPI I/O calls.

**Comment by albenstein** [comment]
I've also checked version 7.5.0 and 7.4.0, I got same error.

**Reply by kursatyurt** [reply]
You are getting the same error because SU2 does not cause it but your openmpi package.

**Reply by albenstein** [reply]
I've also checked with OpenMPI 4.1.5, nothing changed.

**Reply by albenstein** [reply]
I've also checked without mpi like this:
`SU2_CFD config.cfg`
Error is:
`Segmentation fault (core dumped)`

**Reply by kursatyurt** [reply]
Since you have compiled with MPI, you have run the executable with mpirun. Sorry I feel like you are not listening to any advice, I will not spend any time on this topic.

**Reply by albenstein** [reply]
I'm listening and executing your advices, but they are neither helpful nore about error. Although thank you for your time.

**Comment by albenstein** [comment]
I'm attaching SU2 7.5.1 compiling process.
[compile_SU2.txt](https://github.com/su2code/SU2/files/12292036/compile_SU2.txt)

**Comment by albenstein** [comment]
I've just run SU2 in single mode with QuickStart example, it gave no error and running at the moment.
[SU2_single_run.txt](https://github.com/su2code/SU2/files/12292384/SU2_single_run.txt)
